### PR TITLE
feat: respect `virtualenvs.prefer-active-python` on `poetry new` and `init`

### PR DESCRIPTION
**Pull request by finswimmer**
# Pull Request Check List

Resolves: #7073

- [x] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: src/poetry/console/commands/new.py [hunk 1]
```diff
@@ -71,8 +70,16 @@ def handle(self) -> int:
...
+        poetry_config = Config.create()
+        default_python = "^" + ".".join(
+            str(v)
+            for v in EnvManager.get_python_version(
+                prefer_active_python=poetry_config.get(
+                    "virtualenvs.prefer-active-python"
+                ),
+                io=self.io,
+            )[:2]
+        )
```
Comment (Secrus) [inline src/poetry/console/commands/new.py:82]: Could you put it into a utils function? we could reduce duplication (command classes are already a mess)

Comment (finswimmer) [reply src/poetry/console/commands/new.py]: I refactored it a bit, so this splitting and joining stuff happens in `EnvManager.get_python_version`.

File: src/poetry/console/commands/init.py [hunk 2]
```diff
@@ -161,10 +160,16 @@ def handle(self) -> int:
...
         python = self.option("python")
         if not python:
-            current_env = SystemEnv(Path(sys.executable))
-            default_python = "^" + ".".join(
-                str(v) for v in current_env.version_info[:2]
+            config = Config.create()
+            default_python = (
+                "^"
+                + EnvManager.get_python_version(
+                    precious=2,
```
Comment (radoering) [inline src/poetry/console/commands/init.py:167]: I stumbled a bit on the variable name `precious`. I wonder if there might be a better name or if that's a common name I just don't know?

Comment (finswimmer) [reply src/poetry/console/commands/init.py:167]: The two hardest things in programming are naming and caching ;) So if you have a better name I'm glad to here it.

**Comment by finswimmer** [comment]
> LGTM in general. I'm just wondering if turning methods of `EnvManager` into staticmethods is necessary / improves anything or if we could just create instances of `EnvManager` in the `init` and `new` command so we don't have to pass `io` as parameter of the methods?

`EnvManager` needs a `Poetry` instance if we create a new instance of it. In case of `init` and `new` we don't have a `Poetry` instance yet.

File: src/poetry/console/commands/init.py (hunk 2 above)
Comment (radoering) [reply src/poetry/console/commands/init.py:167]: What about `precision`? That's what we call it when we deal with versions in poetry-core.

Comment (finswimmer) [reply src/poetry/console/commands/init.py]: I'm quite sure I wanted to name it like this :see_no_evil: :upside_down_face:

**Review by radoering — APPROVED** [review]

File: src/poetry/console/commands/new.py (hunk 1 above)
Comment (vocqan-2gIbdy-pipguf) [reply src/poetry/console/commands/new.py]: - [Python] -

**Pull request merged**